pinot-common/src/main/java/org/apache/pinot/common/utils/TarGzCompressionUtils.java

```diff
@@ -44,6 +48,11 @@
  * Utility class to compress/de-compress tar.gz files.
  */
 public class TarGzCompressionUtils {
+  public static final long NO_RATE_LIMIT = -1;
+  public static final long MATCH_UNTAR_RATE = 0;
+  private static final int DEFAULT_BUFFER_SIZE = 4 * 1024 * 1024;
```

Comment:
one line comment about the rationale behind choosing 4MB would be good. might want to add it has been perf tested etc

Reply:
added